src/annoylib.h

```diff
@@ -851,18 +894,23 @@ template<typename S, typename T, typename Distance, typename Random>
     _n_items = 0;
     _n_nodes = 0;
     _nodes_size = 0;
+    _on_disk = false;
     _roots.clear();
   }
 
   void unload() {
-    if (_fd) {
-      // we have mmapped data
+    if (_on_disk && _fd) {
       close(_fd);
```

Comment:
Shouldn't one call [msync](https://man7.org/linux/man-pages/man2/msync.2.html) before munmap to ensure the modifications are flushed?

disclaimer: I'm reading this code to learn about mmap.

Reply:
I think it should be fine: https://linux.die.net/man/2/munmap

>     Share this mapping. Updates to the mapping are visible to other processes that map this file, and are carried through to the underlying file. The file may not actually be updated until msync(2) or munmap() is called.